libnmstate/validator.py

```diff
@@ -401,3 +415,36 @@ def _assert_bond_config_is_legal(iface_state, original_iface_state):
                 "Bond option arp_interval is conflicting with miimon, "
                 "please disable one of them by setting to 0"
             )
+
+
+def _validate_slave_ip_config(original_desired_state):
+    """
+    When an interface is created, the slaves/ports cannot have
+    their IP configuration enabled
+    """
+    for iface_state in original_desired_state.interfaces.values():
+        iface_type = iface_state.get(Interface.TYPE)
+        slaves = []
+        if iface_type in SLAVE_GETTERS.keys():
+            slaves = SLAVE_GETTERS[iface_type](iface_state)
+        if slaves:
+            for slave_iface_name in slaves:
+                slave_state = original_desired_state.interfaces.get(
+                    slave_iface_name, {}
+                )
+                if (
+                    slave_state.get(Interface.TYPE)
+                    != InterfaceType.OVS_INTERFACE
```

Comment:
Actaully, OVS interface has two types: `system` and `internal`. The internal interface can have IP enabled while `system` does not.

dedicate tests for these two OVS interface type are also recommended.

Reply:
Maybe I am missing something, but OVS Interface has no type. There is only two kind of OVS Ports "system" and "internal", a port is internal only when it is `ovs-interface` type. So IMO, this check is fine, as it is checking that the port is `OVS_INTERFACE` which means it is "internal" type. Thanks!

Reply:
Please, see [this commit](https://github.com/nmstate/nmstate/pull/525/commits/dcfb27655871972b93c9559a0a3ace410e44d06b). As you can see, the only one condition to be an internal port is to be `ovs-interface` type. If not, it is a system port. Thanks!

Reply:
@elvgarrui @ffmancera The system OVS interface does not listed as `InterfaceType.OVS_INTERFACE`. So this looks good.